framework/web/UrlRule.php

```diff
@@ -216,6 +221,145 @@ public function init()
     }
 
     /**
+     * Initializes normalizer configuration
+     * @param UrlManager $manager the URL manager
+     * @throws \Exception if config is not understood
+     */
+    protected function initNormalizer($manager) {
+        $normalize = $this->normalize;
+
+        $configNoAction = [
+            // whether to collapse multiple slashes to single
+            'collapse-slashes' => false,
+            // remove trailing slash if it exists
+            'remove-trailing-slash' => false,
+            // add trailing slash if there is no one
+            'add-trailing-slash' => false,
+            // what action to perform if pathInfo was changed during normalization
+            // redirect: instant redirect to normalized path
+            // route: return new route (must be configured via 'route')
+            // none: no action, just route by normalized path
+            'action' => 'none',
+            // string: route for $action == 'route', i.e. 'site/redirect'
+            'route' => null
+        ];
+
+        if (is_string($normalize) || is_null($normalize)) {
+            switch ($normalize) {
+                case '':
+                case 'safe-remove-trailing-slash':
```

Comment:
@samdark yes, now I see. 

Ok, let;s imagine this config:

```
'ruleConfig' => [
    'class' => 'yii\web\UrlRule',
    'suffix' => '',
    'normalize' => [
        UrlRule::NORMALIZE_TRAILING_SLASH => false,
        UrlRule::NORMALIZE_COLLAPSE_SLASHES => true
    ]
],
```

does not look good for me.

Maybe Normalizer should be a separate class then? Then it can be configured as usual Yii object. And all settings become class properties (e. g. ```$this->trailingSlash```), then it will be typo-tolerant and IDE-friendly.